### ISSUE TITLE
Investigation on the loadable plugin system through dynamic library

### ISSUE THREAD

**Issue by hyunsik** [hyunsik/tokamak#4]
Even after Tokamak binary is build, we can add external packages to Tokamak. For it, we need to support the plugin loading through dynamic libraries.

This library would be helpful for this work.
https://nagisa.github.io/rust_libloading/libloading/index.html

Currently, I'm thinking that the plugins loaded when the system starts up. I don't intend the plugin system which can be loaded while the system is running. Also, this is an investigation work to make some sample code and a prototype as unit tests. This is because we still don't decide the plugin system design yet.

**Comment by jihoonson** [comment]
I agree with you. Generally, functionality requirements on analytics systems won't change while they are running.
